### PR TITLE
Add freemarker.template.utility.StringUtil tainting support

### PR DESCRIPTION
**Pull request by jandro996**
# What Does This Do

Add tainting support to:

freemarker.template.utility.StringUtil#HTMLEnc
freemarker.template.utility.StringUtil#XMLEnc
freemarker.template.utility.StringUtil#XHTMLEnc
freemarker.template.utility.StringUtil#javaStringEnc
freemarker.template.utility.StringUtil#javaScriptStringEnc
freemarker.template.utility.StringUtil#jsonStringEnc

# Motivation

# Additional Notes

### CODE REVIEW

File: dd-java-agent/instrumentation/freemarker/src/test/groovy/datadog/trace/instrumentation/freemarker/StringUtilCallSiteTest.groovy [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import foo.bar.TestStringUtilSuite
+
+class StringUtilCallSiteTest extends AgentTestRunner {
+
+  @Override
+  protected void configurePreAgent() {
+    injectSysConfig("dd.iast.enabled", "true")
+  }
+
+  void 'test #method'() {
```
Comment (smola) [inline dd-java-agent/instrumentation/freemarker/src/test/groovy/datadog/trace/instrumentation/freemarker/StringUtilCallSiteTest.groovy:16]: Let's add a test checking the behaviour when the input is `null`.

**Review by smola — CHANGES_REQUESTED** [review]

**Review by smola — APPROVED** [review]
